File: inspect_n8n.py
import json
import chardet

def inspect_json():
    print("Reading file to detect encoding...")
    raw_data = open('n8n_response.json', 'rb').read()
    result = chardet.detect(raw_data)
    encoding = result['encoding']
    print(f"Detected encoding: {encoding}")

    try:
        content = raw_data.decode(encoding)
        data = json.loads(content)
        
        print(f"Data type: {type(data)}")
        
        sample = None
        if isinstance(data, list):
            print(f"List length: {len(data)}")
            if len(data) > 0:
                sample = data[0]
        elif isinstance(data, dict):
            print(f"Dict keys: {list(data.keys())}")
            # If it's a wrapper like { results: [...] }
            if 'results' in data:
                print("Found 'results' key")
                if len(data['results']) > 0:
                    sample = data['results'][0]
            else:
                sample = data
        
        if sample:
            print("\n--- SAMPLE ITEM STRUCTURE ---")
            print(json.dumps(sample, indent=2))
            
            # Specific check for Platform
            print("\n--- PLATFORM CHECK ---")
            # Try various paths
            paths = [
                "item.get('platform')",
                "item.get('Platform')",
                "item.get('properties', {}).get('Platform')",
                "item.get('properties', {}).get('Platform', {}).get('select', {}).get('name')"
            ]
            
            item = sample
            for p in paths:
                try:
                    val = eval(p)
                    print(f"{p}: {val}")
                except Exception as e:
                    print(f"{p}: Error {e}")

    except Exception as e:
        print(f"Error parsing JSON: {e}")

if __name__ == "__main__":
    try:
        # Install chardet if missing (unlikely in standard envs but fallback)
        import chardet
        inspect_json()
    except ImportError:
        # Fallback if chardet not available, try utf-16 then utf-8
        print("chardet not found, trying manual encodings")
        try:
            with open('n8n_response.json', 'r', encoding='utf-16') as f:
                print("Reading as utf-16")
                # ... reusing logic would be better but keeping it simple for tool
                data = json.load(f)
                print(json.dumps(data[0] if isinstance(data, list) else data, indent=2))
        except Exception as e:
            print(f"UTF-16 failed: {e}")
            try:
                with open('n8n_response.json', 'r', encoding='utf-8') as f:
                    print("Reading as utf-8")
                    data = json.load(f)
                    print(json.dumps(data[0] if isinstance(data, list) else data, indent=2))
            except Exception as e2:
                print(f"UTF-8 failed: {e2}")

File: inspect_n8n.js
const fs = require('fs');

try {
    const rawData = fs.readFileSync('n8n_response.json');
    let content = '';

    // Force UTF-16LE check
    if (rawData[0] === 0xFF && rawData[1] === 0xFE) {
        console.log('Detected UTF-16LE BOM - Decoding as utf16le');
        content = rawData.toString('utf16le');
    } else {
        console.log('No BOM detected, trying utf8');
        content = rawData.toString('utf8');
    }

    // Clean up content if needed (sometimes BOM char remains at start if not handled perfectly by toString)
    if (content.charCodeAt(0) === 65279) {
        content = content.slice(1);
    }

    const data = JSON.parse(content);
    console.log('Data parsed successfully');

    let item;
    if (Array.isArray(data)) {
        console.log(`Is Array: Yes, Length: ${data.length}`);
        item = data[0];
    } else {
        console.log('Is Object');
        if (data.results) {
            console.log('Has "results" key');
            item = data.results[0];
        } else {
            item = data;
        }
    }

    if (item) {
        console.log('\n--- SAMPLE ITEM ---');
        console.log(JSON.stringify(item, null, 2));
    }

} catch (e) {
    console.error('Error:', e.message);
}
